src/Microsoft.Azure.SignalR/ServerConnections/ClientConnectionContext.cs

```diff
@@ -42,12 +52,19 @@ internal class ClientConnectionContext : ConnectionContext,
         private readonly object _heartbeatLock = new object();
         private List<(Action<object> handler, object state)> _heartbeatHandlers;
 
+        private volatile bool _abortOnClose = true;
+        private volatile Task _lifetimeTask;
+
         public Task CompleteTask => _connectionEndTcs.Task;
 
         public bool IsMigrated { get; }
 
-        // Send "Abort" to service on close except that Service asks SDK to close
-        public bool AbortOnClose { get; set; } = true;
+        // Send "Abort" to service on close except that Service asks SDK to close
+        public bool AbortOnClose
```

Comment:
Why doesn't the auto property work?

Reply:
need to guarantee thread safety as the property can be set and read from multiple threads